lib/trianglify.js

```diff
@@ -65,11 +69,20 @@ function Trianglify(opts) {
       return chroma(opts.color_function(x, y));
     };
   } else {
-    var x_color = chroma.scale(opts.x_colors).mode(opts.color_space);
-    var y_color = chroma.scale(opts.y_colors).mode(opts.color_space);
-    gradient = function(x, y) {
-      return chroma.interpolate(x_color(x), y_color(y), 0.5, opts.color_space);
-    };
+    // Both colors are specified: Interpolate between both.
+    if(opts.x_colors && opts.y_colors){
+      var x_color = chroma.scale(opts.x_colors).mode(opts.color_space);
+      var y_color = chroma.scale(opts.y_colors).mode(opts.color_space);
+      gradient = function(x, y) {
+        return chroma.interpolate(x_color(x), y_color(y), 0.5, opts.color_space);
+      };
+    } else {
+      // One color missing: Use only the specified one.
+      var singleColorScale = chroma.scale(opts.x_colors ? opts.x_colors : opts.y_colors).mode(opts.color_space);
+      gradient = function(x, y) {
+        return singleColorScale(opts.x_colors ? x : y);
+      };
```

Comment:
I'd rather not reference the `opts` in this closure for performance reasons (in some usages of trianglify, the `gradient` function gets run thousands of times per generated image). 

How does this look to you?

```js
var scale = chroma
  .scale(opts.x_colors || opts.y_colors)
  .mode(opts.color_space)

var gradient = opts.x_colors
  ? function(x, y) { return scale(x) }
  : function(x, y) { return scale(y) };
```

Reply:
Makes total sense! I'll fix that ASAP!